src/main/kotlin/io/snyk/plugin/ui/jcef/ApplyFixHandler.kt

```diff
@@ -0,0 +1,138 @@
+                    if (result.isSuccess) {
+                        val script = """
+                            window.receiveApplyFixResponse(true);
+                        """.trimIndent()
+                        jbCefBrowser.cefBrowser.executeJavaScript(script, jbCefBrowser.cefBrowser.url, 0)
+                    } else {
+                        val errorMessage = "Error applying fix: ${result.exceptionOrNull()?.message}"
+                        SnykBalloonNotificationHelper.showError(errorMessage, project)
+                        val errorScript = """
+                            window.receiveApplyFixResponse(false, "$errorMessage");
+                        """.trimIndent()
+                        jbCefBrowser.cefBrowser.executeJavaScript(errorScript, jbCefBrowser.cefBrowser.url, 0)
+                    }
+                }
+            }
+            return@addHandler JBCefJSQuery.Response("success")
+        }
+
+        return object : CefLoadHandlerAdapter() {
+            override fun onLoadEnd(browser: CefBrowser, frame: CefFrame, httpStatusCode: Int) {
+                if (frame.isMain) {
+                    val script = """
+                    (function() {
+                        if (window.applyFixQuery) {
+                            return;
+                        }
+                        window.applyFixQuery = function(value) { ${applyFixQuery.inject("value")} };
+                    })();
+                    """.trimIndent()
+                    browser.executeJavaScript(script, browser.url, 0)
+                }
+            }
+        }
+    }
+
+    private fun applyPatchAndSave(project: Project, filePath: String, patch: String): Result<Unit> {
+        val virtualFile = filePath.toVirtualFile()
+        val patcher = DiffPatcher()
+
+        return try {
+            WriteCommandAction.runWriteCommandAction(project) {
+                val document = FileDocumentManager.getInstance().getDocument(virtualFile)
+                if (document != null) {
+                    val originalContent = document.text
+                    val patchedContent = patcher.applyPatch(originalContent, patcher.parseDiff(patch))
+                    if (originalContent != patchedContent) {
+                        document.setText(patchedContent)
+                    } else {
+                        log("[applyPatchAndSave] Patch did not modify content: $filePath")
+                    }
+                } else {
+                    log("[applyPatchAndSave] Failed to find document for: $filePath")
+                    return@runWriteCommandAction
+                }
+            }
+            Result.success(Unit)
+        } catch (e: Exception) {
+            log("[applyPatchAndSave] Error applying patch to: $filePath. e: $e")
+            Result.failure(e)
+        }
+    }
+
+    private fun log(logMessage: String) {
+        when {
+            enableDebug -> logger.debug(logMessage)
```

Comment:
am not sure if we need this
why don't we directly use `logger<ApplyFixHandler>().trace` ?